lib/presentation/molecules/youtube_player_molecule.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/material.dart';
+import 'package:youtube_player_flutter/youtube_player_flutter.dart';
+
+class YoutubePlayerMolecule extends StatelessWidget {
+  const YoutubePlayerMolecule({
+    required this.url,
+    super.key,
+  });
+  final String url;
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData themeData = Theme.of(context);
+    final ColorScheme colorScheme = themeData.colorScheme;
+    final YoutubePlayerController controller = YoutubePlayerController(
+      initialVideoId: YoutubePlayer.convertUrlToId(url)!,
+      flags: const YoutubePlayerFlags(hideThumbnail: true),
+    );
+    return YoutubePlayerBuilder(
+      player: YoutubePlayer(
+        controller: controller,
+        showVideoProgressIndicator: true,
+        progressIndicatorColor: colorScheme.primary,
```

Comment:
I think this one can be gray so colorScheme.outline.
Check if it looks good and decide, I trust you :)

Reply:
Outline does look better, I've made changes to the progress bar to match this as well